apps/music/js/metadata/flac.js

```diff
@@ -36,36 +37,55 @@ var FLACMetadata = (function() {
     // First four bytes are "fLaC" or we wouldn't be here.
     blobview.seek(4);
 
-    metadata.tag_format = 'flac';
-    return findVorbisCommentBlock(blobview).then(function(block) {
-      if (!block) {
-        return metadata;
+    return new Promise(function(resolve, reject) {
+      metadata.tag_format = 'flac';
+      var has_vorbis_comment = false;
+      var has_picture = false;
+
+      function processBlock(block) {
+        if (!block) {
+          resolve(metadata);
+          return false;
+        }
+
+        if (block.block_type === 4) {
+          readAllComments(block.view, metadata);
+          has_vorbis_comment = true;
+
```

Comment:
Nit: this blank line looks weird to me, but I have no especially strong feelings about it. Just wanted to mention it.